cogs/strike.py

```diff
@@ -698,10 +698,19 @@ async def on_member_update(self, before: discord.Member, after: discord.Member)
         if not after.timed_out or before.timed_out == after.timed_out:
             return
 
+        action: discord.AuditLogAction
+        async for audit_log in main_guild.audit_logs(limit=5):
+            audit_log_action: discord.AuditLogAction = audit_log.action
+            if "auto_moderation_user_communication_disabled" in str(audit_log_action):
+                action = discord.AuditLogAction.auto_moderation_user_communication_disabled
+
+        if not action:
+            action = discord.AuditLogAction.member_update
```

Comment:
I'm a bit confused what you're trying to do here. Set the action type to `auto_moderation_user_communication_disabled` if it was automod that triggered the event? Much more will need to be changed if you pass that action type into the `_confirm_manual_add_strike()` function (E.g. the `MODERATION_ACTIONS` mapping on line 506).

If the above logic is what you were looking for, I think your retrieval of the action type can also be improved:
```python
audit_log_entry: discord.AuditLogEntry
async for audit_log_entry in main_guild.audit_logs(limit=5):
    if audit_log_entry.action == discord.AuditLogAction.auto_moderation_user_communication_disabled:  # noqa: E501
        # noinspection PyArgumentList
        await self._confirm_manual_add_strike(
            strike_user=after,
            action=audit_log_entry.action,
        )
        return

# noinspection PyArgumentList
await self._confirm_manual_add_strike(
    strike_user=after,
    action=discord.AuditLogAction.member_update,
)